[lld] build failure

```
[6/1027] Building CXX object tools/lld/MachO/CMakeFiles/lldMachO.dir/Symbols.cpp.obj
FAILED: tools/lld/MachO/CMakeFiles/lldMachO.dir/Symbols.cpp.obj 
/home/cqwrteur/toolchains/llvm/x86_64-generic-linux-gnu/llvm/bin/clang++ --target=x86_64-windows-gnu --sysroot=/home/cqwrteur/toolchains/llvm/x86_64-windows-gnu/x86_64-windows-gnu -DGTEST_HAS_RTTI=0 -D_FILE_OFFSET_BITS=64 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/home/cqwrteur/toolchains_build/toolchainbuildscripts/llvm/.llvmartifacts/x86_64-windows-gnu/llvm/tools/lld/MachO -I/home/cqwrteur/toolchains_build/llvm-project/lld/MachO -I/home/cqwrteur/toolchains_build/llvm-project/lld/include -I/home/cqwrteur/toolchains_build/toolchainbuildscripts/llvm/.llvmartifacts/x86_64-windows-gnu/llvm/tools/lld/include -I/home/cqwrteur/toolchains_build/toolchainbuildscripts/llvm/.llvmartifacts/x86_64-windows-gnu/llvm/include -I/home/cqwrteur/toolchains_build/llvm-project/llvm/include -I/home/cqwrteur/toolchains_build/llvm-project/llvm/../libunwind/include -rtlib=compiler-rt -fuse-ld=lld -stdlib=libc++ -lc++abi -Wno-unused-command-line-argument -lunwind -stdlib=libc++ -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -ffunction-sections -fdata-sections -flto=thin -O3 -DNDEBUG -std=c++17  -fno-exceptions -funwind-tables -fno-rtti -MD -MT tools/lld/MachO/CMakeFiles/lldMachO.dir/Symbols.cpp.obj -MF tools/lld/MachO/CMakeFiles/lldMachO.dir/Symbols.cpp.obj.d -o tools/lld/MachO/CMakeFiles/lldMachO.dir/Symbols.cpp.obj -c /home/cqwrteur/toolchains_build/llvm-project/lld/MachO/Symbols.cpp
/home/cqwrteur/toolchains_build/llvm-project/lld/MachO/Symbols.cpp:24:15: error: static assertion failed due to requirement 'sizeof(void *) != 8 || sizeof(lld::macho::Defined) == 88': Try to minimize Defined's size; we create many instances
   24 | static_assert(sizeof(void *) != 8 || sizeof(Defined) == 88,
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[15/1027] Linking CXX shared library bin/libLLVMX86CodeGen.dll
ninja: build stopped: subcommand failed.
```

--- a/lld/MachO/Symbols.h
+++ b/lld/MachO/Symbols.h
@@ -14,6 +14,7 @@
 #include "Target.h"
 
 #include "llvm/Object/Archive.h"
+#include "llvm/Support/Compiler.h"
 #include "llvm/Support/MathExtras.h"
 
 namespace lld {
@@ -152,7 +153,8 @@ public:
   // Whether this symbol should appear in the output symbol table.
   bool includeInSymtab : 1;
   // The ICF folding kind of this symbol: None / Body / Thunk.
-  ICFFoldKind identicalCodeFoldingKind : 2;
+  LLVM_PREFERRED_TYPE(ICFFoldKind)
+  uint8_t identicalCodeFoldingKind : 2;
   // Symbols marked referencedDynamically won't be removed from the output's
   // symbol table by tools like strip. In theory, this could be set on arbitrary
   // symbols in input object files. In practice, it's used solely for the

